sklearn/decomposition/tests/test_truncated_svd.py

```diff
@@ -177,24 +186,27 @@ def test_singular_values():
 
     apca = TruncatedSVD(n_components=2, algorithm='arpack',
                         random_state=rng).fit(X)
-    rpca = TruncatedSVD(n_components=2, algorithm='arpack',
-                        random_state=rng).fit(X)
-    assert_array_almost_equal(apca.singular_values_, rpca.singular_values_, 12)
+    pca = TruncatedSVD(n_components=2, algorithm='randomized',
+                       random_state=rng).fit(X)
+    assert_allclose(apca.singular_values_, pca.singular_values_, rtol=1e-2)
```

Comment:
This fails with a higher relative tolerance which I assume means that singular_values are very tiny..

Reply:
No they are not tiny,
```py
>>> apca.singular_values_
[17.57446934 17.47000919]
>>> pca.singular_values_
[17.52238412 17.3596697 ]
```
but also certainly not equal up to 1e-12 retlative tolerance. 

This was previously passing because we were comparing arpack solver against itself by mistake a few lines above.